### PR TITLE
chore: Upgrade deps

### PR DESCRIPTION
**Pull request by pomykaczi**
- upgrade nodejs 
- changed cdkVersion and constructsVersion
- removed lodash dependency

### CODE REVIEW

File: src/utils/find.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Account, EnvRegExp } from '../project';
+
+export function findAccountTypeByEnvironment(obj: Record<string, Account>, predicate: string): string | undefined {
+  for (const key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      const list = obj[key].environments?.filter((environment) => new EnvRegExp(environment).test(predicate));
```
Comment (mpiltz) [inline src/utils/find.ts:6]: Could this be written using `some` instead of `filter`?

**Review by mpiltz — COMMENTED** [review]
Just a minor note :)

File: src/utils/find.ts (hunk 1 above)
Comment (pomykaczi) [reply src/utils/find.ts:6]: Actually it can ... thank you to pointed that out.

File: src/utils/find.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Account, EnvRegExp } from '../project';
+
+export function findAccountTypeByEnvironment(obj: Record<string, Account>, predicate: string): string | undefined {
+  for (const key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      if (obj[key].environments?.some((environment) => new EnvRegExp(environment).test(predicate))) {
+        return key;
+      }
+    }
+  }
```
Comment (aripalo) [inline src/utils/find.ts:10]: Just a matter of preference: I would probably use `Object.entries` for this myself:
```ts
Object
  .entries(obj)
  .filter(([key, value]) => value.environments?.some((environment) => new RegExp(environment).test(predicate)))
  .map(([key]) => key).at(0);
```

The benefit of this is that there's no need to use `hasOwnProperty` check as `Object.entries` only returns object's own properties.

But again, this is just more about _personal taste_ than actual "code quality".

File: src/utils/isSet.ts [hunk 3]
```diff
@@ -1,6 +1,4 @@
-import { values } from 'lodash';
 
-
-export function isSet(value?: string | undefined): boolean {
-  return (typeof value === 'string' && values.length > 0);
-}
\ No newline at end of file
+export function isNonEmptyString(value?: unknown): value is string {
```
Comment (aripalo) [inline src/utils/isSet.ts:2]: rename the file from ~~`isSet.ts`~~ to `isNonEmptyString.ts`, so it matches the implementation and doesn't confuse people later.

File: src/utils/find.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Account, EnvRegExp } from '../project';
+
+export function findAccountTypeByEnvironment(obj: Record<string, Account>, predicate: string): string | undefined {
```
Comment (aripalo) [inline src/utils/find.ts:3]: I'd like to see a simple unit test for this. You can create a new file `find.test.ts` into the same folder and Jest should pick it up automatically.

Not that we had much unit tests before :) But let's use the "boy scout rule" of _leave the place in better condition you found it_.

File: src/utils/get.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+export function get(obj: Record<string, any> | undefined, path: string, defValue?: any): any {
```
Comment (aripalo) [inline src/utils/get.ts:2]: I'd like to see a simple unit test for this. You can create a new file `find.test.ts` into the same folder and Jest should pick it up automatically.

Not that we had much unit tests before :) But let's use the "boy scout rule" of _leave the place in better condition you found it_.

**Review by aripalo — CHANGES_REQUESTED** [review]
Minor fixes/additions like file renaming, couple extra tests and a suggestion about one implementation.

File: src/utils/find.ts (hunk 2 above)
Comment (pomykaczi) [reply src/utils/find.ts:10]: I have changed my preference. I like the less`for` approach.

**Review by aripalo — APPROVED** [review]

**Pull request merged**